sparse/impl/KokkosSparse_spiluk_numeric_impl.hpp

```diff
@@ -21,644 +21,1000 @@
-            if (ipos != -1) {
-              if (col < rowid)
-                L_values(ipos) += lxu;
-              else
-                U_values(ipos) += lxu;
-            }
-          });  // end for kk
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, k1, k2),
+                           [&](const size_type k) {
+                             const auto col = Base::L_entries(k);
+                             Base::lset(k, 0.0);
+                             Base::iw(my_team, col) = k;
+                           });
+
+      team.team_barrier();
+
+      k1 = Base::U_row_map(rowid);
+      k2 = Base::U_row_map(rowid + 1);
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, k1, k2),
+                           [&](const size_type k) {
+                             const auto col = Base::U_entries(k);
+                             Base::uset(k, 0.0);
+                             Base::iw(my_team, col) = k;
+                           });
+
+      team.team_barrier();
+
+      // Unpack the ith row of A
+      k1 = Base::A_row_map(rowid);
+      k2 = Base::A_row_map(rowid + 1);
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, k1, k2),
+                           [&](const size_type k) {
+                             const auto col  = Base::A_entries(k);
+                             const auto ipos = Base::iw(my_team, col);
+                             if (col < rowid) {
+                               Base::lset(ipos, Base::aget(k));
+                             } else {
+                               Base::uset(ipos, Base::aget(k));
+                             }
+                           });
 
       team.team_barrier();
-    }  // end for k
 
-    // if (my_thread == 0) {
-    Kokkos::single(Kokkos::PerTeam(team), [&]() {
-      nnz_lno_t ipos = iw(my_team, rowid);
+      // Eliminate prev rows
+      k1 = Base::L_row_map(rowid);
 #ifdef KEEP_DIAG
-      if (U_values(ipos) == 0.0) {
-        U_values(ipos) = 1e6;
-      }
+      k2 = Base::L_row_map(rowid + 1) - 1;
 #else
-      if (U_values(ipos) == 0.0) {
-        U_values(ipos) = 1e6;
-      } else {
-        U_values(ipos) = 1.0 / U_values(ipos);
-      }
+      k2           = Base::L_row_map(rowid + 1);
+#endif
+      for (auto k = k1; k < k2; k++) {
+        const auto prev_row = Base::L_entries(k);
+        const auto udiag    = Base::uget(Base::U_row_map(prev_row));
+#ifdef KEEP_DIAG
+        Base::divide(team, Base::lget(k), udiag);
+#else
+        fact = Base::multiply(team, 1.0, fact, udiag);
+#endif
+        auto fact = Base::lget(k);
+        Kokkos::parallel_for(
+            Kokkos::TeamThreadRange(team, Base::U_row_map(prev_row) + 1,
+                                    Base::U_row_map(prev_row + 1)),
+            [&](const size_type kk) {
+              const auto col  = Base::U_entries(kk);
+              const auto ipos = Base::iw(my_team, col);
+              if (ipos != -1) {
+                scalar_t buff[100];  // 10*10
```

Comment:
How do you choose ```100``` for ```buff```? In the case of unblocked spiluk, do we still need to allocate ```buff```?

Reply:
I just picked a max block size, 10. I have changed the code to use buff_size=0 for unblocked.